store/persistent/storethehash/storethehash.go

```diff
@@ -0,0 +1,136 @@
+package storethehash
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/filecoin-project/storetheindex/store"
+	cidprimary "github.com/ipld/go-storethehash/store/primary/cid"
+
+	"github.com/ipfs/go-cid"
+	sth "github.com/ipld/go-storethehash/store"
+	peer "github.com/libp2p/go-libp2p-core/peer"
+)
+
+var _ store.Storage = &sthStorage{}
+
+// TODO: Benchmark and fine-tune for better performance.
+const DefaultIndexSizeBits = uint8(24)
+const DefaultBurstRate = 4 * 1024 * 1024
+const DefaultSyncInterval = time.Second
+
+type sthStorage struct {
+	dir   string
+	store *sth.Store
+}
+
+func New(dir string) (store.Storage, error) {
```

Comment:
IMO, it is preferable to return a pointer to the concrete type (`*sthStorage`) here.  That way the caller can decide what interface they want to cast it to if there are multiple interfaces that it may satisfy.

In general, [accept interfaces, return types](https://mycodesmells.com/post/accept-interfaces-return-struct-in-go)

Reply:
It makes total sense! Thank you for the reference, I wasn't aware of this proverb :)